Lib/glyphsLib/builder/transformations/intermediate_layers.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import uuid
+
+from fontTools.varLib.models import VariationModel, normalizeValue
+
+from glyphsLib.classes import GSLayer, GSNode, GSPath
+
+
+logger = logging.getLogger(__name__)
+
+
+def resolve_intermediate_components(font):
+    components_with_intermediate_layers = set()
+    all_intermediate_locations = set()
+    for glyph in font.glyphs:
+        for layer in glyph.layers:
+            if "coordinates" in layer.attributes:
+                # First, let's find glyphs with intermediate layers
+                # which have components which don't have intermediate layers
+                for shape in layer.components:
+                    check_component_has_sparse_layer(font, shape, layer)
+                # Later we will check if everyone who uses me as a component
+                # has the same intermediate layers as I do
+                components_with_intermediate_layers.add(glyph.name)
+                all_intermediate_locations.add(tuple(layer.attributes["coordinates"]))
+
+
+def simple_variation_model(font):
+    tags = [axis.axisTag for axis in font.axes]
+    locations = [x.axes for x in font.masters]
+    limits = {tag: (min(x), max(x)) for tag, x in zip(tags, (zip(*locations)))}
+    master_locations = []
+    for loc in locations:
+        this_loc = {}
+        for ix, axisTag in enumerate(tags):
+            axismin, axismax = limits[axisTag]
+            this_loc[axisTag] = normalizeValue(loc[ix], (axismin, axismin, axismax))
```

Comment:
I'm not sure this "simple_variation_model" is correct in the general: you're discarding the actual _default_ master axis coordinates (which can be set by the user via 'Variable Font Origin' and glyphsLib should know since it uses it when creating the DS axes) and instead assuming that default == min all the time; the choice of a default is not neutral, it may have consequences throughout the designspace because in the variation model that's always the first thing that doesn't move and is used as reference point for computing the deltas for the rest of the masters thus will influence how the VF behaves everywhere.

Reply:
Right, but I don't care about OT deltas here. I'm *only* using this model for interpolating the outlines between masters. I don't need it to be correct in the general case, which is why it's "simple".

Reply:
You're right, and I was only dragging my feet because I thought it was going to be a pain to fix, but it turned out not to be.